--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -1884,7 +1884,10 @@
 
 		s->addSaveFunc([this, window, devSwitchScript, optionsDeviceSwitch, selectedDevice] {
 			if (optionsDeviceSwitch->changed()) {
-				Utils::Platform::runSystemCommand(devSwitchScript + " " + optionsDeviceSwitch->getSelected(), "", nullptr);
+				if (Utils::Platform::runSystemCommand(devSwitchScript + " " + optionsDeviceSwitch->getSelected(), "", nullptr) != 0) {
+					window->pushGui(new GuiMsgBox(window, _("DEVICE SWITCH FAILED"), _("OK"), nullptr));
+					return;
+				}
 				window->pushGui(new GuiMsgBox(window, _("Device will be switched on next reboot"),
 					_("Reboot now"), [] { Utils::Platform::quitES(Utils::Platform::QuitMode::REBOOT); },
 					_("later"), nullptr)
